src/ort_genai_c.h

```diff
@@ -218,9 +225,8 @@ OGA_EXPORT const int32_t* OGA_API_CALL OgaGenerator_GetSequence(const OgaGenerat
 
 OGA_EXPORT OgaResult* OGA_API_CALL OgaCreateTokenizer(const OgaModel* model, OgaTokenizer** out);
 OGA_EXPORT void OGA_API_CALL OgaDestroyTokenizer(OgaTokenizer*);
-OGA_EXPORT OgaResult* OGA_API_CALL OgaTokenizerEncodeBatch(const OgaTokenizer*, const char** strings, size_t count, OgaSequences** out);
```

Comment:
We should have a simple/faster common case function vs requiring the complex batching functions. Since this change requires copying every string before tokenizing, and using another object, we should also have a low level function like:

`OgaTokenizerEncode(const OgaTokenizer*, const char* string, OgaSequences** out);`

It's a common case to just pass in one string, so we should make this simple and efficient. It also would pair with OgaTokenizerDecode.

Or perhaps use the OgaBuffer. The batching APIs shouldn't be less efficient than the non batching ones.

Reply:
I added a new function for single encode.

Reply:
I can also go back to using the previous signature if you feel that would be better. It will just make the c# code quite unpleasant with all the Marshalling to make it work with const char** and const char*** types.

Reply:
Since we control both sides, what would be your ideal C API to interface with C# for this part?

The C API should ideally allow all language bindings to be as efficient as possible, so I want to avoid forced copies. Do you know if Java/Rust would need to do similar here?

Reply:
I don't know the answer to this question. If we don't claim backwards compatibility, we can change this going forward. But i think we should try to make a reasonable decision now to avoid problems for end users.

Reply:
Let's have both for now, and call the new ones OgaTokenizer*BatchStrings.

Should we call OgaStringArray just OgaStrings? Since we don't say OgaSequenceArray 